pyfr/plugins/solutionwriter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolutionWriterPlugin(BasePlugin, H5Writer):
+    name = 'solutionwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, *args, **kwargs):
+        BasePlugin.__init__(self, intg, cfgsect)
```

Comment:
In your above example why are you calling `super().__init(...)` for classes that inherit from `object`?  This will attempt to call the constructor for `object` which takes no parameters with a bunch of parameters, hence the error.